packages/realm-web/src/tests/utils/MockNetworkTransport.ts

```diff
@@ -47,22 +49,55 @@ export class MockNetworkTransport implements NetworkTransport {
     }
 
     /** @inheritdoc */
-    fetchAndParse<RequestBody extends any, ResponseBody extends any>(
+    fetch<RequestBody extends any>(
         request: Request<RequestBody>,
-    ): Promise<ResponseBody> {
+    ): Promise<FetchResponse> {
         if (!request.headers || Object.keys(request.headers).length === 0) {
             delete request.headers;
         }
-        if (!request.body) {
+        // Save a parsed body, instead of a string
+        const { body } = request;
```

Comment:
I wouldn't use destructuring here and just use `request.body` in the two places below. It is somewhat misleading since if that branch is taken `body` will no longer be the same as `request.body` and it would be a bug to use `body` later on in the function

Reply:
I agree - good catch.